src/ntp/server.c

```diff
@@ -238,9 +238,15 @@ static void request_process_loop(const int fd, const char *ipstr, const int prot
 	while (true)
 	{
 		unsigned char buf[48];
-		struct sockaddr src_addr;
+		struct sockaddr_storage src_addr;
```

Comment:
Using `struct sockaddr_storage` seems to be the right thing here. Interesting that the `man recvfrom` didn't mention this when I looked at it while implementing this. Anyway, [socket.h description](https://pubs.opengroup.org/onlinepubs/009696799/basedefs/sys/socket.h.html) says:

> The header shall define the `sockaddr_storage` structure. This structure shall be:
>
> - Large enough to accommodate all supported protocol-specific address structures
> - Aligned at an appropriate boundary so that pointers to it can be cast as pointers to protocol-specific address structures and used to access the fields of those structures without alignment problems